app/src/main/java/com/insudev/euvictodo/buisnessLogic/UseCase.kt

```diff
@@ -0,0 +1,115 @@
+package com.insudev.euvictodo.buisnessLogic
+
+import com.insudev.euvictodo.models.Filters
+import com.insudev.euvictodo.models.ModelInterface
+import com.insudev.euvictodo.models.Sorting
+import com.insudev.euvictodo.models.TodoModel
+import com.insudev.euvictodo.mvi.TodoListChangeResult
+import io.reactivex.Observable
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+
+class UseCase {
+
+    private val retrofit = Retrofit.Builder()
+        .baseUrl("http://euvictodoapi.herokuapp.com/")
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.createAsync())
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+
+    private val service = retrofit.create<TodoService>(TodoService::class.java)
+
+    private val api = TodoRetroService(service)
+
+    companion object {
+        val instance = UseCase()
+    }
+
+    fun addTodo(content: String, prevId: Int): Observable<TodoListChangeResult> {
+        return Observable.just(
+            TodoListChangeResult.Completed(
+                TodoModel(
+                    prevId, System.currentTimeMillis() / 1000, content,
+                    false, arrayListOf("")
+                ), null
+            ) as TodoListChangeResult
+        )
+            .startWith(TodoListChangeResult.Pending())
+            .onErrorReturnItem(TodoListChangeResult.Error("Błąd dodawnaia notatki"))
+    }
+
+    fun sync(toSync: ArrayList<TodoModel>): Observable<TodoListChangeResult> {
+        return api.sync(toSync)
+            .map {
+                if (it) TodoListChangeResult.Completed("Zsynchronizowano", null)
```

Comment:
nawet w takich apkach powinniśmy unikać języka polskiego, potem niepotrzebnie wejdzie nam w krew i to wyjdzie